### PR TITLE
spec: add the glossary

### PR DESCRIPTION
**Pull request by nickolaev**
Signed-off-by: Nikolay Nikolaev <nnikolay@vmware.com>

## Description
Adding the NSM Glossary as specified by the Documentation Call.

## How Has This Been Tested?

- [ ] Covered by existing integration testing
- [ ] Added integration testing to cover
- [ ] Tested locally
- [ ] Have not tested

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by nickolaev** [comment]
@jeffsaelens could you peek here?

File: docs/spec/glossary.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+
+**Network Service Endpoint** - a container, POD, virtual machine or physical forwarder with knowledge of network resources.  A network service endpoint accepts zero or more connection requests from clients which want to receive the Network Service it is offering.
+
+**Network Service Client** - a requester or consumer of the Network Service.
+
+**Network service registry** -  A cluster, VIM or physical network level registry of NSM components
+
+**Network Service Registry Domain** - The registry of all network services, network service endpoint(s) providing the services, and network service managers registered to a specific NSR
+
+**Network Service Manager Domain** - The collective of Network Service Client(s) and Network Service Endpoint(s) directly managed by an NSMgr.
```
Comment (edwarnicke) [inline docs/spec/glossary.md]: This should probably come after the definition of Network Service Manager.

Comment (edwarnicke) [reply docs/spec/glossary.md]: Don't want to hold this up over a nit... would be good to fix with follow up PR.

Comment (jeffsaelens) [reply docs/spec/glossary.md]: agreed. This one is on me, the shuffling got a little out of order.

**Comment by jeffsaelens** [comment]
@nickolaev looks good. Just need to rearrange the section Ed called out.

**Pull request merged**